<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>trf &mdash; documantation 1 documentation</title>
    
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="documantation 1 documentation" href="index.html" />
    <link rel="up" title="Repeat masking" href="repeat-masking.html" />
    <link rel="prev" title="Repeat masking" href="repeat-masking.html" />
   
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="trf">
<h1>trf<a class="headerlink" href="#trf" title="Permalink to this headline">¶</a></h1>
<p>Tandem Repeats Finder <a class="footnote-reference" href="#id2" id="id1">[1]</a> (<a class="reference external" href="http://tandem.bu.edu/trf/trf.unix.help.html">link</a>)</p>
<p><em>version 4.07b</em></p>
<div class="section" id="program-parameters">
<h2>Program parameters<a class="headerlink" href="#program-parameters" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash"><div class="highlight"><pre><span></span>$ /exports/software/trf/trf-4.07b/trf

Tandem Repeats Finder, Version 4.07b
Copyright <span class="o">(</span>C<span class="o">)</span> Dr. Gary Benson 1999-2012. All rights reserved.


Please use: /exports/software/trf/trf-4.07b/trf File Match Mismatch Delta PM PI Minscore MaxPeriod <span class="o">[</span>options<span class="o">]</span>

Where: <span class="o">(</span>all weights, penalties, and scores are positive<span class="o">)</span>
  <span class="nv">File</span> <span class="o">=</span> sequences input file
  <span class="nv">Match</span>  <span class="o">=</span> matching weight
  <span class="nv">Mismatch</span>  <span class="o">=</span> mismatching penalty
  <span class="nv">Delta</span> <span class="o">=</span> indel penalty
  <span class="nv">PM</span> <span class="o">=</span> match probability <span class="o">(</span>whole number<span class="o">)</span>
  <span class="nv">PI</span> <span class="o">=</span> indel probability <span class="o">(</span>whole number<span class="o">)</span>
  <span class="nv">Minscore</span> <span class="o">=</span> minimum alignment score to report
  <span class="nv">MaxPeriod</span> <span class="o">=</span> maximum period size to report
  <span class="o">[</span>options<span class="o">]</span> <span class="o">=</span> one or more of the following
               -m    masked sequence file
               -f    flanking sequence
               -d    data file
               -h    suppress html output
               -r    no redundancy elimination
               -ngs  more compact .dat output on multisequence files, returns <span class="m">0</span> on success. You may pipe input in with this option using - <span class="k">for</span> file name. Short <span class="m">50</span> flanks are appended to .dat output. See more information on TRF Unix Help web page.
</pre></div>
</div>
<p><a class="reference external" href="http://tandem.bu.edu/trf/trf.unix.help.html">Recommended usage</a>
(plus html output suppression)</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>trf yoursequence.txt <span class="m">2</span> <span class="m">7</span> <span class="m">7</span> <span class="m">80</span> <span class="m">10</span> <span class="m">50</span> <span class="m">500</span> -f -d -m -h
</pre></div>
</div>
</div>
<div class="section" id="program-outputs">
<h2>Program outputs<a class="headerlink" href="#program-outputs" title="Permalink to this headline">¶</a></h2>
<p>Outputs are written to input directory, using input filename suffixed by
parameter values and file type</p>
<div class="section" id="infile-2-7-7-80-10-50-500-mask">
<h3>infile.2.7.7.80.10.50.500.mask<a class="headerlink" href="#infile-2-7-7-80-10-50-500-mask" title="Permalink to this headline">¶</a></h3>
<p>Hard-masked fasta format sequence file (option <code class="docutils literal"><span class="pre">-m</span></code>):</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>&gt;seq1
GTTGTTTTTCTGAGACCGAAAAGGCTTGTTTTTATGTAGCAAGTTCAAGAACTGGTTTCG
GTATCGCAGGGTATGTATGGTCTGAGACGTATAACTGTCACGANNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNTATGATGAATGGGTAATTTTTAAAGTATCTCACATAACTTTTAGATG
TAGTAAGACCATGTTTGCGAATCGAATCTCTTACTGGTTGAACGCCAAAGGACCTTCCAT
...
&gt;seq2
...
</pre></div>
</div>
</div>
<div class="section" id="infile-2-7-7-80-10-50-500-dat">
<h3>infile.2.7.7.80.10.50.500.dat<a class="headerlink" href="#infile-2-7-7-80-10-50-500-dat" title="Permalink to this headline">¶</a></h3>
<p>Text file with space delimited blocks (option <code class="docutils literal"><span class="pre">-d</span></code>):</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>Tandem Repeats Finder Program written by:

Gary Benson
Program in Bioinformatics
Boston University
Version 4.07b


Sequence: seq1



Parameters: 2 7 7 80 10 50 500


3404 3433 14 2.1 14 93 0 51 30 16 30 23 1.96 AGCAGCACTTGAGT AGCAGTACTTGAGTAGCAGCACTTGAGTAG
6344 6383 18 2.2 18 95 0 71 45 2 10 42 1.51 ATTATTATAAGGAATTAC ATTATTATAAGGAATTATATTATTATAAGGAATTACATTA
...
6237875 6237915 16 2.6 16 96 3 75 0 4 48 46 1.23 TGTGTGTGTGTGCGTG TGTGTGTGTGTGCGTGTGTGTGTGTGTGCGTGTGTTGTGTG


Sequence: seq2



Parameters: 2 7 7 80 10 50 500


15030 15056 14 1.9 14 100 0 54 66 0 0 33 0.92 TTAAATAAATAAAT TTAAATAAATAAATTTAAATAAATAAA
...
</pre></div>
</div>
<p>Each block contains:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span>cols 1+2:  Indices of the repeat relative to the start of the sequence
col 3:     Period size of the repeat
col 4:     Number of copies aligned with the consensus pattern
col 5:     Size of consensus pattern (may differ slightly from the period size)
col 6:     Percent of matches between adjacent copies overall
col 7:     Percent of indels between adjacent copies overall
col 8:     Alignment score
cols 9-12: Percent composition for each of the four nucleotides
col 13:    Entropy measure based on percent composition
col 14:    Consensus sequence
col 15:    Repeat sequence
</pre></div>
</div>
</div>
</div>
<div class="section" id="grid-engine-wrapper">
<h2>Grid engine wrapper<a class="headerlink" href="#grid-engine-wrapper" title="Permalink to this headline">¶</a></h2>
<p>trf.sh:</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>

<span class="c1">#$ -V</span>
<span class="c1">#$ -cwd</span>
<span class="c1">#$ -j y</span>
<span class="c1">#$ -o $JOB_ID.log</span>

<span class="c1"># define $DATADIR and $SEQFILE before/when calling this script, e.g.:</span>
<span class="c1">#       export DATADIR=`pwd`</span>
<span class="c1">#       qsub -v SEQFILE=seq.fa ...</span>

<span class="nv">WORKDIR</span><span class="o">=</span>/scratch/$USER/trf
mkdir -p $WORKDIR
cp <span class="nv">$DA</span>TADIR/$SEQFILE $WORKDIR
<span class="nb">cd</span> $WORKDIR
nice -n <span class="m">10</span> /exports/software/trf/trf-4.07b/trf $SEQFILE <span class="m">2</span> <span class="m">7</span> <span class="m">7</span> <span class="m">80</span> <span class="m">10</span> <span class="m">50</span> <span class="m">500</span> -f -d -m -h
rsync -a --remove-source-files ./<span class="si">${</span><span class="nv">SEQFILE</span><span class="si">}</span>* <span class="nv">$DA</span>TADIR
</pre></div>
</div>
</div>
<div class="section" id="qsub-command">
<h2>qsub command<a class="headerlink" href="#qsub-command" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nb">export</span> <span class="nv">DATADIR</span><span class="o">=</span><span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>
qsub -v <span class="nv">SEQFILE</span><span class="o">=</span>seq.fa /path/to/trf.sh
</pre></div>
</div>
</div>
<div class="section" id="parallel-command">
<h2>parallel command<a class="headerlink" href="#parallel-command" title="Permalink to this headline">¶</a></h2>
<p>Split large sequence file into blocks with pyfasta, process each block
using parallel to submit to gridengine with one slot per job,
then combine outputs into single <code class="docutils literal"><span class="pre">.mask</span></code> and <code class="docutils literal"><span class="pre">.dat</span></code> files.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>pip install pyfasta
pip install numpy
</pre></div>
</div>
<div class="highlight-bash"><div class="highlight"><pre><span></span><span class="nb">export</span> <span class="nv">DATADIR</span><span class="o">=</span><span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>
<span class="nb">export</span> <span class="nv">INFILE</span><span class="o">=</span>seq <span class="c1"># OMIT EXTENSION!</span>
<span class="nb">export</span> <span class="nv">SCRIPTDIR</span><span class="o">=</span>~
pyfasta split -n <span class="m">32</span> $INFILE
parallel -j <span class="m">32</span> qsub -q nice.q -v <span class="nv">SEQFILE</span><span class="o">={}</span> -sync y $SCRIPTDIR/trf.sh ::: $INFILE.??.fa
cat $INFILE.??.fa.*.mask &gt; $INFILE.fa.mask
cat $INFILE.??.fa.*.dat &gt; $INFILE.fa.dat
</pre></div>
</div>
<div class="highlight-bash"><div class="highlight"><pre><span></span>rm *.log
rm $INFILE.??.fa.*.mask
rm $INFILE.??.fa.*.dat
rm $INFILE.??.fa
</pre></div>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<table class="docutils footnote" frame="void" id="id2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td>Benson G (1999)
Tandem repeats finder: a program to analyze DNA sequences
<em>Nucleic Acids Research</em>
<strong>27</strong>, 573-580.</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">trf</a><ul>
<li><a class="reference internal" href="#program-parameters">Program parameters</a></li>
<li><a class="reference internal" href="#program-outputs">Program outputs</a><ul>
<li><a class="reference internal" href="#infile-2-7-7-80-10-50-500-mask">infile.2.7.7.80.10.50.500.mask</a></li>
<li><a class="reference internal" href="#infile-2-7-7-80-10-50-500-dat">infile.2.7.7.80.10.50.500.dat</a></li>
</ul>
</li>
<li><a class="reference internal" href="#grid-engine-wrapper">Grid engine wrapper</a></li>
<li><a class="reference internal" href="#qsub-command">qsub command</a></li>
<li><a class="reference internal" href="#parallel-command">parallel command</a></li>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="repeat-masking.html">Repeat masking</a><ul>
      <li>Previous: <a href="repeat-masking.html" title="previous chapter">Repeat masking</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/trf.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2016, Blaxter Lab, University of Edinburgh.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.3.5</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.7</a>
      
      |
      <a href="_sources/trf.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>